packages/create-cli/src/lib/utils.ts

```diff
@@ -0,0 +1,93 @@
+import { readFile, rm, stat, writeFile } from 'node:fs/promises';
+import {
+  NX_JSON_CONTENT,
+  NX_JSON_FILENAME,
+  PROJECT_JSON_CONTENT,
+  PROJECT_JSON_FILENAME,
+  PROJECT_NAME,
+} from './constants';
+
+export type SetupResult = {
+  filename: string;
+  teardown: boolean;
+};
+
+export async function setupFile(
+  filename: string,
+  content = '',
+): Promise<SetupResult> {
+  const setupResult: SetupResult = {
+    filename,
+    teardown: false,
+  };
+
+  try {
+    const stats = await stat(filename);
+    if (!stats.isFile()) {
+      await writeFile(filename, content);
+    }
+  } catch (error) {
+    if ((error as Error).message.includes('no such file or directory')) {
```

Comment:
Safer handling:

```suggestion
    if (error instanceof Error && error.message.includes('no such file or directory')) {
```

Reply:
Thanks for this improvement! Way better!